### PR TITLE
Fix saving contents of webview notebooks when closing a notebook

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #4779

### CODE REVIEW

File: src/client/datascience/interactive-ipynb/nativeEditor.ts [hunk 1]
```diff
@@ -291,10 +296,11 @@ export class NativeEditor extends InteractiveBase implements INotebookEditor {
         await this.loadPromise;
         return super.show(preserveFocus);
     }
-    public dispose(): Promise<void> {
+    public async dispose(): Promise<void> {
+        this.isDisposing = true;
         super.dispose();
+        await this.close();
         this.model?.dispose(); // NOSONAR
```
Comment (DonJayamanne) [inline src/client/datascience/interactive-ipynb/nativeEditor.ts:303]: Model was being disposed too early, even before the prompt is displayed.

File: src/client/datascience/interactive-ipynb/nativeEditor.ts [hunk 2]
```diff
@@ -241,6 +242,10 @@ export class NativeEditor extends InteractiveBase implements INotebookEditor {
 
     @captureTelemetry(Telemetry.SyncAllCells)
     public async syncAllCells(): Promise<void> {
+        // Possible webview has been closed (and we'retrying to save just as user closed the webview editor).
+        if (this.isDisposing) {
```
Comment (DonJayamanne) [inline src/client/datascience/interactive-ipynb/nativeEditor.ts:246]: Webview doesn't exist, hence no way to synchronize anything.

File: src/client/datascience/interactive-ipynb/nativeEditor.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/client/datascience/interactive-ipynb/nativeEditor.ts:303]: Hence model is disposed last.

**Review by rchiodo — APPROVED** [review]
Your title seems a bit off, but otherwise looks good.

**Comment by IanMatthewHuff** [comment]
Yeah, I'm not sure what's up with the title. I'd change before you submit just so we can find it later.

**Review by IanMatthewHuff — APPROVED** [review]

**Review by DavidKutu — APPROVED** [review]

**Comment by DonJayamanne** [comment]
Took me 15 minutes to find this PR, lol, didn't realize the title was wrong.

**Pull request merged**